sw/host/tests/manuf/cp_provision_init/src/main.rs

```diff
@@ -120,8 +120,40 @@ fn load_and_run_sram_program(opts: &Opts, transport: &TransportWrapper) -> Resul
     // Once the SRAM program has printed a message over the console, we can continue with an LC
     // transition to mission mode, initiated on the host side.
     let _ = UartConsole::wait_for(&*uart, r"CP provisioning end.", opts.timeout)?;
+    jtag.disconnect()?;
+
+    Ok(())
+}
+
+fn reset_and_lock(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    // Set the TAP straps for the lifecycle controller and reset.
+    transport
+        .pin_strapping("PINMUX_TAP_LC")?
+        .apply()
+        .context("failed to apply LC TAP strapping")?;
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset")?;
 
-    // TODO(#19453): transition to mission mode.
+    // Connect to the LC TAP via JTAG.
+    let jtag = opts.init.jtag_params.create(transport)?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to LC TAP over JTAG")?;
+
+    trigger_lc_transition(
+        transport,
+        jtag.clone(),
+        DifLcCtrlState::TestLocked0,
+        None,
+        true,
+        opts.init.bootstrap.options.reset_delay,
```

Comment:
If #19526 goes it first, you'll need to add a parameter to `trigger_lc_transition`. Is it safe to reconnect the JTAG at this point? It's not clear to me if ROM execution is enabled after the transition (I think not?). Could you add a comment to indicate that?